File: content/python/templates/prime.py
# Prime Number Check

def is_prime(n):
    """Check if n is a prime number."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

# Test
for num in range(1, 20):
    if is_prime(num):
        print(f"{num} is prime")

File: content/java/templates/06_Algorithms/prime.java
// Prime Number Check

public class Prime {
    public static boolean isPrime(int n) {
        if (n < 2) return false;
        if (n == 2) return true;
        if (n % 2 == 0) return false;
        
        for (int i = 3; i <= Math.sqrt(n); i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }
    
    public static void main(String[] args) {
        System.out.print("Prime numbers from 1 to 20: ");
        for (int num = 1; num <= 20; num++) {
            if (isPrime(num)) {
                System.out.print(num + " ");
            }
        }
        System.out.println();
    }
}
